commodore/compile.py

```diff
@@ -59,41 +64,36 @@ def _fetch_customer_config(cfg, customer_id):
     cfg.register_config('customer', repo)
 
 
-def _regular_setup(config, cluster_id):
+def _regular_setup(config, cluster_id, target):
     try:
         cluster = fetch_cluster(config, cluster_id)
     except ApiError as e:
         raise click.ClickException(f"While fetching cluster specification: {e}") from e
-    customer_id = cluster['tenant']
 
-    target_name = update_target(config, cluster)
-    if target_name != 'cluster':
-        raise click.ClickException(
-            f"Only target with name 'cluster' is supported, got {target_name}")
+    update_target(config, target)
+    update_params(cluster, target)
 
     # Fetch components and config
     _fetch_global_config(config, cluster)
-    _fetch_customer_config(config, customer_id)
+    _fetch_customer_config(config, cluster['tenant'])
     fetch_components(config)
-    update_target(config, cluster)
-    # Fetch catalog
-    catalog_repo = fetch_customer_catalog(config, cluster['gitRepo'])
+    update_target(config, target)
 
-    return cluster, target_name, catalog_repo
+    # Fetch catalog
+    return fetch_customer_catalog(config, cluster['gitRepo'])
 
 
-def _local_setup(config, cluster_id):
+def _local_setup(config, cluster_id, target):
     click.secho('Running in local mode', bold=True)
     click.echo(' > Will use existing inventory, dependencies, and catalog')
+
     # Currently, a target name other than "cluster" is not supported
-    target_name = 'cluster'
-    target_yml = P('inventory/targets') / f"{target_name}.yml"
-    if not target_yml.is_file():
-        raise click.ClickException(f"Invalid target: {target_name}")
-    click.echo(f" > Using target: {target_name}")
+    if not target_file(target).is_file():
+        raise click.ClickException(f"Invalid target: {target}")
+    click.echo(f" > Using target: {target}")
 
     click.echo(' > Reconstructing Cluster API data from target')
-    cluster = reconstruct_api_response(target_yml)
+    cluster = reconstruct_api_response(target)
```

Comment:
Since it looks like we only need the cluster and tenant IDs anymore, can we change `reconstruct_api_response()` to only get these two values?

Reply:
Refactored to `read_cluster_and_tenant(target: str) -> Tuple[str, str]`.